core/src/test/java/com/linecorp/armeria/common/HttpRequestBuilderTest.java

```diff
@@ -120,16 +122,198 @@ void buildWithPathParams2() {
                                             .build())
                 .isInstanceOf(IllegalStateException.class);
 
-        request = HttpRequest.builder().get("/{foo}/{bar}/:id/:/{/foo/}/::/a{")
-                             .pathParams(ImmutableMap.of("id", 3, "bar", 2, "foo", 1, "", 4, "/foo/", 5))
-                             .pathParam(":", 6)
+        request = HttpRequest.builder().get("/{foo}/{bar}/:id/{/foo/}/::/a{")
+                             .pathParams(ImmutableMap.of("id", 3, "bar", 2, "foo", 1, "/foo/", 4))
+                             .pathParam(":", 5)
                              .build();
-        assertThat(request.path()).isEqualTo("/1/2/3/4/5/6/a{");
+        assertThat(request.path()).isEqualTo("/1/2/3/4/5/a{");
+    }
 
-        request = HttpRequest.builder().get("/{}/:")
-                             .pathParams(ImmutableMap.of("", "foo"))
-                             .build();
-        assertThat(request.path()).isEqualTo("/foo/foo");
+    @Test
+    void shouldRejectEmptyPathParams() {
+        assertThatThrownBy(() -> HttpRequest.builder().pathParam("", "foo"))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("empty");
+
+        assertThatThrownBy(() -> HttpRequest.builder().pathParams(ImmutableMap.of("", "foo")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("empty");
+    }
+
+    @ParameterizedTest
+    @CsvSource({ "true", "false" })
+    void shouldFailOnMissingPathParam(boolean curly) {
+        assertThatThrownBy(() -> HttpRequest.builder()
+                                            .get(curly ? "/{foo}" : "/:foo")
+                                            .build())
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("param 'foo'");
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "https://host:8080,      https://host:8080",
+            "https://host:8080/,     https://host:8080/",
+            "https://host:8080/foo,  https://host:8080/foo",
+            "https://host:8080/foo/, https://host:8080/foo/",
+            "https://host:8080/:,    https://host:8080/:",
+            "https://host:8080/:/,   https://host:8080/:/",
+            "https://host:8080/:/:,  https://host:8080/:/:",
+            "https://host:8080/:/:/, https://host:8080/:/:/",
+            // Note: We don't test '{}' because absolute URIs don't accept '{' or '}'.
```

Comment:
Could you also add a test for this case?
Could you also add the reference of it, please?

Reply:
This is actually not the case anymore. Let me update the test case.